wrangler-service/src/main/java/co/cask/wrangler/service/s3/S3Handler.java

```diff
@@ -98,21 +100,12 @@ public void testS3Connection(HttpServiceRequest request, HttpServiceResponder re
     try {
       // Extract the body of the request and transform it to the Connection object.
       RequestExtractor extractor = new RequestExtractor(request);
-      Connection connection = extractor.getContent(Charsets.UTF_8.name(), Connection.class);
-      if (connection == null) {
-        responder.sendError(400, "Connection information is missing from the request body.");
-        return;
-      }
-      ConnectionType connectionType = ConnectionType.fromString(connection.getType().getType());
-      if (connectionType == ConnectionType.UNDEFINED || connectionType != ConnectionType.S3) {
```

Comment:
does ui uses `ConnectionType.UNDEFINED` for anything?

Reply:
no, because it would just hit this clause and error out. It seems like this was intended to check for the case where the user provides an invalid type like "S4", but was implemented incorrectly. I have another PR that consolidates error handling and will make sure this scenario is properly handled.